timemachine/fe/rbfe.py

```diff
@@ -274,9 +277,57 @@ def setup_initial_states(
     return initial_states
 
 
+def rebalance_lambda_schedule(
+    initial_states: Sequence[InitialState],
+    setup_initial_state_fn: Callable[[float], InitialState],
+    trajectories: Sequence[Trajectory],
+    target_overlap: float,
+    xtol: float = 1e-4,
+) -> Sequence[InitialState]:
+    assert 0.0 < target_overlap <= 1.0
+    assert len(initial_states) == len(trajectories)
+    initial_lambs = np.array([state.lamb for state in initial_states])
+
+    lambda_min = min(initial_lambs)
+    lambda_max = max(initial_lambs)
+
+    u_kn, n_k = compute_u_kn(trajectories, initial_states)
+
+    f_k = MBAR(u_kn, n_k).f_k
+
+    overlap_dist = make_fast_approx_overlap_distance_fxn(initial_lambs, u_kn, f_k, n_k)
+    target_dist = 1.0 - target_overlap
+
+    greedy_prot = greedily_optimize_protocol(
+        overlap_dist, target_dist, bisection_xtol=xtol, protocol_interval=(lambda_min, lambda_max)
+    )
+    new_schedule = np.asarray(greedy_prot)
+    if len(greedy_prot) > len(initial_lambs):
+        warnings.warn("Optimized schedule has more windows than initial schedule, falling back to initial schedule")
+        new_schedule = initial_lambs
+    else:
+        print(
+            f"Optimized schedule has {len(new_schedule)} windows compared to {len(initial_lambs)} windows initially, target overlap {target_overlap}"
+        )
```

Comment:
Maybe just warn, rather than override, in case optimized schedule has more windows than initial schedule? There may be cases where this is intentional. (e.g. if `target_overlap > bisection_min_overlap`)

Something like:
```suggestion
    if len(greedy_prot) > len(initial_lambs):
        warnings.warn("Optimized schedule has more windows than initial schedule, falling back to initial schedule")    
    
    print(
        f"Optimized schedule has {len(new_schedule)} windows compared to {len(initial_lambs)} windows initially, target overlap {target_overlap}"
    )
```
(may have messed up whitespace in GH comment editor)

Reply:
> There may be cases where this is intentional. (e.g. if target_overlap > bisection_min_overlap)

I am not necessarily opposed to just warning and returning the new protocol (though do we want the number of windows to be greater than 48?), but the code prevents the scenario you describe in: https://github.com/proteneer/timemachine/blob/6c5358077b0c3cadff718892eb5868b542e6dbd0/timemachine/fe/rbfe.py#L800-L802 

I enforce that the bisection `overlap >= target_overlap` based on previous discussions, though would be opening to removing this restriction.